ffs/scheduler/scheduler.go

```diff
@@ -62,6 +63,9 @@ type Scheduler struct {
 	ctx      context.Context
 	cancel   context.CancelFunc
 	finished chan struct{}
+
+	jobStatusCache     map[ffs.APIID]map[cid.Cid]map[cid.Cid]ffs.FilStorage
```

Comment:
With maintain this cache of in process deal state. It maps FFS instance id > data cid > proposal cid > `FilStorage` state (which now includes the `Status`). This cache can be queried to build responses about cid storage state or job state.